idea/idea-core/src/org/jetbrains/kotlin/idea/core/overrideImplement/OverrideImplementMembersHandler.kt

```diff
@@ -106,7 +117,59 @@ abstract class OverrideImplementMembersHandler : LanguageCodeInsightActionHandle
                 selectedElements: Collection<OverrideMemberChooserObject>,
                 copyDoc: Boolean
         ) {
-            insertMembersAfter(editor, classOrObject, selectedElements.map { it.generateMember(classOrObject, copyDoc) })
+            val selectedMemberDescriptors = selectedElements.associate { it.generateMember(classOrObject, copyDoc) to it.descriptor }
+
+            val classBody = classOrObject.body
+            if (classBody == null) {
+                insertMembersAfter(editor, classOrObject, selectedMemberDescriptors.keys)
+                return
+            } else {
+                val offset = editor?.caretModel?.offset ?: classBody.startOffset
+                val offsetCursorElement = PsiTreeUtil.findFirstParent(classBody.containingFile.findElementAt(offset)) {
+                    it.parent == classBody
+                }
+                if (offsetCursorElement != null && offsetCursorElement != classBody.rBrace) {
+                    insertMembersAfter(editor, classOrObject, selectedMemberDescriptors.keys)
+                    return
+                }
+            }
+            val classLeftBrace = classBody.lBrace
+
+            val allSuperMemberDescriptors = selectedMemberDescriptors.values
+                .mapNotNull { it.containingDeclaration as? LazyClassDescriptor }
+                .associate { it to it.declaredCallableMembers.toList() }
+
+            val implementedElements = mutableMapOf<CallableMemberDescriptor, KtDeclaration>()
+
+            fun ClassDescriptor.findElement(memberDescriptor: CallableMemberDescriptor): KtDeclaration? {
+                return implementedElements[memberDescriptor]
+                    ?: (findCallableMemberBySignature(memberDescriptor)?.source?.getPsi() as? KtDeclaration)?.also {
+                        implementedElements[memberDescriptor] = it
+                    }
+            }
+
+            fun getAnchor(selectedElement: KtDeclaration): PsiElement? {
+                val lastElement = classOrObject.declarations.lastOrNull()
```

Comment:
You should return `null` by default for case:
```kotlin
class Other {
    fun test() {
        val a = 1
    }
<caret>

    fun otherTest() {

    }

}
```

Reply:
This function will not be executed because it is returned here:
https://github.com/t-kameyama/kotlin/blob/9fb07118db46f84598a87326bc2e6410ab3a93d4/idea/idea-core/src/org/jetbrains/kotlin/idea/core/overrideImplement/OverrideImplementMembersHandler.kt#L133-L136